src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,47 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int DEFAULT_RADIUS = 10;
+    private static final int NUMBERS = 5;
+    private static final String COLOR = Color.WHITE.name();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+    private final Random random = new Random();
+
+    public Figure getRandomFigure() {
+        int choice = random.nextInt(NUMBERS);
+        switch (choice) {
+            case 0:
+                int squareSide = random.nextInt(10) + 1;
+                return new Square(squareSide, colorSupplier.getRandomColor());
+            case 1:
+                int rectLength = random.nextInt(10) + 1;
+                int rectWidth = random.nextInt(10) + 1;
+                return new Rectangle(rectLength, rectWidth, colorSupplier.getRandomColor());
+            case 2:
+                int triFirstLeg = random.nextInt(10) + 1;
+                int triSecondLeg = random.nextInt(10) + 1;
+                return new RightTriangle(triFirstLeg, triSecondLeg, colorSupplier.getRandomColor());
+            case 3:
+                int circleRadius = random.nextInt(10) + 1;
+                return new Circle(circleRadius, colorSupplier.getRandomColor());
+            case 4:
+                return prepareIsoscelesTrapezoid();
+            default:
+                return getDefaultFigure();
+        }
+    }
+
+    private IsoscelesTrapezoid prepareIsoscelesTrapezoid() {
```

Comment:
All private methods should be below public methods
and you can do it for each Figure